src/main/scala/io/opentargets/etl/backend/drug/Drug.scala

```diff
@@ -116,7 +103,9 @@ object Drug extends Serializable with LazyLogging {
 
     val dataframesToSave: Map[String, (DataFrame, IOResourceConfig)] = Map(
       "drug" -> (drugDf, outputs.drug),
-      "mechanism_of_action" -> (mechanismOfActionProcessedDf, outputs.mechanismOfAction),
+      // coalesce as otherwise the ~5k mechanisms are spread over 200 files.
+      "mechanism_of_action" -> (mechanismOfActionProcessedDf
+        .coalesce(10), outputs.mechanismOfAction),
```

Comment:
@JarrodBaker coalesce is not really a good action here is was thought to produce files for other use. I would not use personally any shuffle partition number reduction as when you join with any other dataset on the right when the number of partition will be 10 and therefore any number of segments to load into memory will be 10 and the number of cores to use parallise will be reduced. Also, coalesce disables shuffling so you could end up with a uneven (see [1](https://sparkbyexamples.com/spark/spark-repartition-vs-coalesce/)) number of elements distributed across partitions.

when the number of partitions want to be reduced but keeping the data evenly distributed (or at least as mush as possible) some repartition family function should be used. `repartition` with a lower partition number and some column ordering or you want a even distribution but not sorted in partition then `repartitionByRange` when more complex distribution or `repartitionByExpression`.

That said, I would recomend **not** to reshape the number or partitions. you can always cat all or load the dataset properly. I would no expect to use the output manually but rather using the proper tools.

Reply:
so if you turgther want to explore or keep a file with all the MOAs with custom select fields then use the dataset and produce your own small file coalesced from your custom query